experiments/AMIP/user_io/ci_plots.jl

```diff
@@ -0,0 +1,203 @@
+# this file follows the ClimaAtmos.Diagnostics and ci_plots interfaces
+import CairoMakie
+import CairoMakie.Makie
+import ClimaAnalysis
+import ClimaAnalysis: Visualize as viz
+import ClimaAnalysis: SimDir, slice_time, slice
+import ClimaAnalysis.Utils: kwargs as ca_kwargs
+import ClimaCoreSpectra: power_spectrum_2d
+
+using Poppler_jll: pdfunite
+import Base.Filesystem
+
+const LARGE_NUM = typemax(Int)
+const LAST_SNAP = LARGE_NUM
+const FIRST_SNAP = -LARGE_NUM
+const BOTTOM_LVL = -LARGE_NUM
+const TOP_LVL = LARGE_NUM
+
+function Makie.get_tickvalues(yticks::Int, ymin, ymax)
+    return range(max(ymin, 0), ymax, yticks)
+end
+
+YLINEARSCALE = Dict(:axis => ca_kwargs(dim_on_y = true, yticks = 10, ytickformat = "{:.3e}"))
+
+long_name(var) = var.attributes["long_name"]
+short_name(var) = var.attributes["short_name"]
+
+"""
+    make_plots_generic(
+        output_path::Union{<:AbstractString, Vector{<:AbstractString}},
+        plot_path,
+        vars,
+        args...;
+        plot_fn = nothing,
+        output_name = "summary",
+        summary_files = String[],
+        MAX_NUM_COLS = 1,
+        MAX_NUM_ROWS = min(4, length(vars)),
+        kwargs...,
+    )
+Create plots for each variable in `vars` and save them to `output_path`. The number of plots per
+page is determined by `MAX_NUM_COLS` and `MAX_NUM_ROWS`. The `plot_fn` function is used to create the
+plots. If `plot_fn` is not provided, a default plotting function is used. The default plotting function
+is determined by the keyword arguments `kwargs`.
+"""
+function make_plots_generic(
+    output_path::Union{<:AbstractString, Vector{<:AbstractString}},
+    plot_path,
+    vars,
+    args...;
+    plot_fn = nothing,
+    output_name = "summary",
+    summary_files = String[],
+    MAX_NUM_COLS = 1,
+    MAX_NUM_ROWS = min(4, length(vars)),
+    kwargs...,
+)
+    # When output_path is a Vector with multiple elements, this means that this function is
+    # being used to produce a comparison plot. In that case, we modify the output name, and
+    # the number of columns (to match how many simulations we are comparing).
+    is_comparison = output_path isa Vector
+    #
+    # However, we don't want to do this when the vector only contains one element.
+    if is_comparison && length(output_path) == 1
+        # Fallback to the "output_path isa String" case
+        output_path = output_path[1]
+        is_comparison = false
+    end
+
+    if is_comparison
+        MAX_NUM_COLS = length(output_path)
+        save_path = output_path[1]
+        output_name *= "_comparison"
+    else
+        save_path = plot_path
```

Comment:
Do we need both `output_path` and `plot_path` in this function? It seems like we could set `save_path = output_path` here, but I could be missing something :)

Reply:
Ah, output path is where the files are saved, plot path is for plots. We can rename `output_path` to `file_path` to clarify this.